src/hotspot/share/runtime/arguments.hpp

```diff
@@ -148,6 +148,15 @@ class SystemProperty : public PathString {
 // Helper class for controlling the lifetime of JavaVMInitArgs objects.
 class ScopedVMInitArgs;
 
+// Arguments passed from JavaMain via the property CRaCJavaMainArgs
```

Comment:
> Duplicating structure definition and relying on the same ABI is too fragile in my opinion.

If you mean `JavaMainArgs` could change its content without us noticing, then maybe it's better that we define our own struct. Otherwise, duplicating structure definition is actually the same thing as defining it in a header file and then including the header twice, since `#include` in C++ is literally copy-and-paste.